src/Stratis.Bitcoin/Utilities/FullNodeExtensions.cs

```diff
@@ -68,28 +69,30 @@ public static async Task RunAsync(this IFullNode node)
         /// <param name="shutdownCompleteMessage">Message to display on the console when the shutdown is complete.</param>
         public static async Task RunAsync(this IFullNode node, CancellationToken cancellationToken, string shutdownMessage, string shutdownCompleteMessage)
         {
-            node.Start();
-
-            if (!string.IsNullOrEmpty(shutdownMessage))
-            {
-                Console.WriteLine();
-                Console.WriteLine(shutdownMessage);
-                Console.WriteLine();
-            }
+            var nodeLifetime = node.Services.ServiceProvider.GetRequiredService<INodeLifetime>() as NodeLifetime;
```

Comment:
Why not `node.NodeLifetime`?

Reply:
`node.NodeLifeTime` would be `null` at this point - i.e. before `Start` is called.

Reply:
Might be a good idea to comment that?

Reply:
Done.